### PR TITLE
Card: Add `isCompact` prop and `Overline` sub-component

### PR DESCRIPTION
**Pull request by JoaoSilvaGrafana**
**What is this feature?**

This adds two new concepts to the `Card` component.

An `isCompact` prop, that makes the padding of the `Card` tighter:
| Normal | Compact |
| --- | --- |
| Screenshot 2024-02-07 at 12 23 08 | Screenshot 2024-02-07 at 12 23 13 |

An `Overline` component that can be composed inside `Card`:
Screenshot 2024-02-07 at 12 24 22

**Why do we need this feature?**

To keep the component in sync with the designs

**Who is this feature for?**

Anyone who uses the `Card` component

**Which issue(s) does this PR fix?**:

Fixes part of #75548

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by Clarity-89 — APPROVED** [review]

File: packages/grafana-ui/src/components/Card/Card.tsx [hunk 1]
```diff
@@ -47,15 +51,24 @@ const CardContext = React.createContext<{
...
  */
-export const Card: CardInterface = ({ disabled, href, onClick, children, isSelected, className, ...htmlProps }) => {
+export const Card: CardInterface = ({
+  disabled,
+  href,
+  onClick,
+  children,
+  isSelected,
+  isCompact,
+  className,
```
Comment (eledobleefe) [inline packages/grafana-ui/src/components/Card/Card.tsx:61]: This `classname` is there due to `CardConteiner`, isn't it? So it's deprecated as that component, no?

**Review by eledobleefe — COMMENTED** [review]
Would it be a good idea to add the **Overline Card** to the card documentation?
Btw, there seems to be some issue with the documentation styles in the examples:
Captura de pantalla 2024-02-09 a las 11 01 45

File: packages/grafana-ui/src/components/Card/Card.tsx (hunk 1 above)
Comment (JoaoSilvaGrafana) [reply packages/grafana-ui/src/components/Card/Card.tsx:61]: I guess so, I didnt make changes to that, just added another prop 🤷

**Comment by JoaoSilvaGrafana** [comment]
> Would it be a good idea to add the **Overline Card** to the card documentation? Btw, there seems to be some issue with the documentation styles in the examples: Captura de pantalla 2024-02-09 a las 11 01 45

@eledobleefe I have added a section for the Overline in the documentation. The other issue you found seems to be related to how the `Preview` component works in the new storybook. It adds `p` tags to all text that is not wrapped in something. I don't really have time now to go figure out what we could use instead of `Preview` so I created an [issue](https://github.com/grafana/grafana/issues/82248) for this to fix later 👍

**Review by eledobleefe — APPROVED** [review]
LGTM 👌

**Pull request merged**